### PR TITLE
Create config file on CLI action with user prompt for analytics

### LINKED ISSUE #2843: Interactively turn on pendo at first run
* On OSS the first time they run a command we ask them if we can turn on "analytics" to improve our product with a link to our documentation outlining what we collect.
* When they say "yes" we save that response in a config file somewhere along with a unique ID.
* On Enterprise, this shouldn't ask - it should just turn it on if it's the first run.

If the config file doesn't exist, it's assumed that means it's the first run.

The link to our documentation page (provided by @josefaworks ):
https://docs.gitops.weave.works/docs/feedback-and-telemetry/

### PR DESCRIPTION
**Pull request by joshri**
Part of #2843 

all commands now attempt to read the config file before running, and create a config file if it doesn't exist - with a user prompt for analytics!

### CODE REVIEW

File: cmd/gitops/root/cmd.go
Comment (opudrovs) [inline cmd/gitops/root/cmd.go:107]: ❔ Should not the value of `Analytics` be set by default to false in a new config? @ozamosi WDYT, is this assignment to false needed here? Can we keep just:

```
if err == nil {
    gitopsConfig.Analytics = true
}
```

File: pkg/config/config.go
Comment (opudrovs) [inline pkg/config/config.go:70]: ⚠️ Unneeded blank line, can be removed.

File: pkg/config/config.go
Comment (opudrovs) [inline pkg/config/config.go:68]: ⚠️ Unneeded blank line, can be removed.

File: cmd/gitops/root/cmd.go
Comment (opudrovs) [inline cmd/gitops/root/cmd.go:96]: 💡  Do we need a space after the ellipsis? And should we capitalize `w`? At least it will be read easier? For some reason, in other log messages we usually add a space before the ellipsis too, maybe it's a convention in CLI?

File: cmd/gitops/root/cmd.go
Comment (opudrovs) [inline cmd/gitops/root/cmd.go:94]: ⚠️ Can this empty line be removed? We don't usually use them after `if err != nil {`.

**Comment by opudrovs** [comment]
❔ Will the following acceptance criteria be implemented solely in the enterprise repo?

"On Enterprise, this shouldn't ask - it should just turn it on if it's the first run."

The PR description says it closes the ticket completely (same issue as with my config commands PR).

**Review by opudrovs — APPROVED** [review]
Tested it, works as expected, great job! 🎉

**Comment by joshri** [comment]
> ❔ Will the following acceptance criteria be implemented solely in the enterprise repo?
> 
> "On Enterprise, this shouldn't ask - it should just turn it on if it's the first run."
> 
> The PR description says it closes the ticket completely (same issue as with my config commands PR).

Yep - going to open a follow-up PR in enterprise after yours is merged!

**Comment by ozamosi** [comment]
@JamWils or @josefaworks: the ticket for this talked about linking to our documentation - I think that's this page, but it needs updating for pendo, and I'm not sure if it requires any legalese given that this is now a 3rd party: https://docs.gitops.weave.works/docs/feedback-and-telemetry/

Should we hold off on merging this until the documentation is there?

**Comment by opudrovs** [comment]
@ozamosi @joshri yes, turns out it's the same URL (with updates):
https://docs.gitops.weave.works/docs/feedback-and-telemetry/

**Comment by opudrovs** [comment]
Picking up this PR, because we need to merge it before other Pendo PRs I suppose. Will just add the link to the docs and will create an enterprise counterpart for it.

**Comment by opudrovs** [comment]
@ozamosi 
- Refactored the analytics config initialization by explicitly setting default values.
- Added text with a link to the feedback and telemetry documentation page before the prompt. Updated the pre-link text and prompt based on @ozamosi 's suggestion.
- Reworked config imports for consistency within this PR and with the Pendo CLI events PR (and to make all config package aliases lowercase).

**Comment by opudrovs** [comment]
Please re-review.

File: cmd/gitops/root/cmd.go
Comment (opudrovs) [inline cmd/gitops/root/cmd.go:122]: @ozamosi and should we exit on a save config error here?

Comment (opudrovs) [reply cmd/gitops/root/cmd.go]: @ozamosi just changed this code to `_ = config.SaveConfig(gitopsConfig)` to ignore errors on config save silently here.

**Comment by opudrovs** [comment]
Hurrah! Will create the enterprise counterpart PR for it today.

**Pull request merged**
